client/my-sites/checkout/src/components/is-for-business-checkbox.tsx

```diff
@@ -0,0 +1,86 @@
+import { FormStatus, useFormStatus } from '@automattic/composite-checkout';
+import { useShoppingCart, convertTaxLocationToLocationUpdate } from '@automattic/shopping-cart';
+import { hasCheckoutVersion, styled } from '@automattic/wpcom-checkout';
+import { CheckboxControl } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import InlineSupportLink from 'calypso/components/inline-support-link';
+import useCartKey from '../../use-cart-key';
+const CheckboxWrapper = styled.div`
+	margin-top: 16px;
+
+	#checkout-is-business-checkbox input[type='checkbox']:checked {
+		background: ${ ( props ) => props.theme.colors.primary };
+		border-color: ${ ( props ) => props.theme.colors.primary };
+	}
+
+	a.inline-support-link#checkout-is-business-checkbox__link {
+		color: ${ ( props ) => props.theme.colors.primary };
+	}
+`;
+export function IsForBusinessCheckbox() {
+	const translate = useTranslate();
+	const { formStatus } = useFormStatus();
+
+	const cartKey = useCartKey();
+	const { responseCart, updateLocation, isLoading, isPendingUpdate } = useShoppingCart( cartKey );
+
+	const isUnitedStateWithBusinessOption = ( () => {
+		if ( responseCart.tax.location.country_code !== 'US' ) {
+			return false;
+		}
+		const zipCode = parseInt( responseCart.tax.location.postal_code ?? '0', 10 );
+		if ( zipCode >= 43000 && zipCode <= 45999 ) {
+			// Ohio; OH
+			return true;
+		}
+		if ( ( zipCode >= 6000 && zipCode <= 6389 ) || ( zipCode >= 6391 && zipCode <= 6999 ) ) {
+			// Connecticut; CT
+			return true;
+		}
+		return false;
+	} )();
+
+	const isChecked = responseCart.tax.location.is_for_business ?? false;
+	const isDisabled = formStatus !== FormStatus.READY || isLoading || isPendingUpdate;
+
+	if ( ! isUnitedStateWithBusinessOption || ! hasCheckoutVersion( 'business-use-tax' ) ) {
+		return null;
+	}
+
+	return (
+		<CheckboxWrapper>
+			<CheckboxControl
+				id="checkout-is-business-checkbox"
+				label={
+					translate(
+						'Is this purchase for business? {{link}}Learn more.{{/link}}',
+						{
+							components: {
+								link: (
+									<InlineSupportLink
+										id="checkout-is-business-checkbox"
+										supportContext="tax-exempt-customers"
+										showIcon={ false }
+									/>
+								),
+							},
+						}
+						// As far as I can tell, label will correctly render the
```

Comment:
Yes, you're correct. The label is rendered here and it will work for any component:

https://github.com/WordPress/gutenberg/blob/2b67cd9a4f37170c1e6c87b14788933c1ffb03c5/packages/components/src/checkbox-control/index.tsx#L143

That means the types for this component are incorrect:

https://github.com/WordPress/gutenberg/blob/2b67cd9a4f37170c1e6c87b14788933c1ffb03c5/packages/components/src/checkbox-control/types.ts#L24

The type should probably be `ReactNode` instead of `string`. Apart from this PR you might consider making a PR on the gutenberg repo to fix that.

Reply:
This comment was brought over with the label from the [AssignToAllPaymentMethods](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/src/payment-methods/credit-card/assign-to-all-payment-methods.tsx#L64-L65) which was added by @noahtallen some time ago.

> The type should probably be ReactNode instead of string. Apart from this PR you might consider making a PR on the gutenberg repo to fix that.

Adding a note to do this 📝 Thanks!

Reply:
Oh yeah! I guess I brought up the same comment back then (with fewer details): https://github.com/Automattic/wp-calypso/pull/78711/files#r1260433004 Anyway, clearly it hasn't been fixed.